### DISCUSSION TITLE
Spacing between collumns is converted to 1 space

### DISCUSSION THREAD

**Discussion by EntiusGJ** [jsvine/pdfplumber#417] [Ask for help with specific PDFs]
Good day,
I am new to PdfPlumber but got one application working. Started on the next but this one is tabular text but the space between the columns is only 1 space. The text in the one of the columns contains also spaces and therefore impossible to distinct the separate columns. 
Is PDFPlumber recognizing tabular formatted lines? Or can I enable this. Is there a option to separate the columns by a Tab character or other non printable one?
This is an example of the body of an invoice:
![image](https://user-images.githubusercontent.com/49858691/115143472-d749eb00-a047-11eb-851b-409e716f8494.png)
This is the text:
04015400264613 PAMPERS ACTIVE JUMBO PACK MINI 94'S 02 185.00 185.00 2 321.74 370.00
06009900265322 MOSTRA DI CAFE C/BEAN 1KG, FORZA 11 162.88 162.88 2 141.63 162.88

any suggestions are welcome :-)
Thanks in advance,
Gerard

**Comment by samkit-jain** [comment]
Hi @EntiusGJ Appreciate your interest in the library. Request you to share the PDF so that I can look into this in more detail. Before sharing the PDF, please ensure that you redact any sensitive information from it.

Looking at the screenshot, I will recommend 2 options:
1. Use the table extraction strategy as
    ```python
    {
        "horizontal_strategy": "lines",
        "vertical_strategy": "text"
    }
    ```
    You may also try with `"vertical_strategy": "lines"` in case there are any hidden vertical separators in the table.
2. Use the table extraction strategy as
    ```python
    {
        "horizontal_strategy": "lines",
        "vertical_strategy": "explicit",
        "explicit_vertical_lines": []  # < List of X coordinates of the vertical separators
    }
    ```
    in case the position of the vertical separators in the table never changes.

Have a look at https://github.com/jsvine/pdfplumber#table-extraction-methods to know more on how to use pdfplumber for table extraction.

**Comment by EntiusGJ** [comment]
Good day to you all 😊

 

Sorry for the delay but been busy lately.

 

I am fairly new to Python and not into the syntax, I am asking help again.

My simple program is this and try to incorporate the suggested strategy.

 

 ts = {"vertical_strategy": "lines", "horizontal_strategy": "text"}

    PdfFilename = os.path.basename(InFilename)

    (file, ext) = os.path.splitext(PdfFilename)

    #username = input("Hit any key......")

    with open(OutFilename, 'w') as f:

        f.write('....... \n' )

        f.write('')

        with pdfplumber.open(InFilename) as pdf :

            pages = pdf.pages 

            for page in pdf.pages:

                #print(page.number)

                text = page.extract_text (table_settings=ts)

                #print (text)

                f.write(text)    

            # end for pdf.Pages

        #end with PDF file

 

When adding the ‘ts’ variable the program executes the 2 f.write commands and bails out.

 

Any suggestions how to tackle this?

 

Thanks in advance and stay healthy.

 

Met vriendelijke groet / With kind regards 

 

Gerard Entius

 

 

Versterstraat 36 | 9301 Bloemfontein |  * +27 (0) 798 723 122

**Reply by samkit-jain** [reply]
Hi Gerard, congrats to you and wish you good luck on your journey with Python. In order to help you further, request you to share the PDF in question. Please ensure that you redact any sensitive information from it.

**Comment by EntiusGJ** [comment]
Hi,

 

Sorry fort his late reply, but it is project with no high priority.

 

Attached is the PDF I want to decode and store the data in a database. I Also attached the text as I can retrieve is now. The first and last  line I added for testing purposes and the personal info is masked with “?”.

 

Thanks for your help and please let me know hot to get a improved ( TAB separated? ) output.

 

Stay healthy 😊 

 

 

Met vriendelijke groet / With kind regards 

 

Gerard Entius

 

 

Versterstraat 36 | 9301 Bloemfontein |  * +27 (0) 798 723 122
